### DISCUSSION TITLE
[V5 TESTING] How can I use HttpContextContract attributes (e.g. auth etc.) in JAPA test files?

### DISCUSSION THREAD

**Discussion by remidosol** [adonisjs/core#2067] [Help] [answered]
Hello there, I'm trying to write test with **AdonisJS v5**. 
~In v4.1~, I **used to be able to** pass a lot of parameters that also used with **HttpContextContract now** (e.g. auth, request, response). 
But I have a lot of issue now. 

For example, how can I use **auth** parameter for using **_login()_** function or how can I get **_tokenHash_** for asserting?
Thanks already.

@thetutlage 
@RomainLanz

**Comment by thetutlage** [comment, marked as answer]
Japa is just a test runner, so it has zero knowledge of the framework. In v4, AdonisJS provided a proper testing environment and hence it was easier to login a user inside a test.

v5 doesn't have a testing environment YET. So you have to dig a little deeper. Unfortunately, I don't have any help to provide right now, coz I myself have to go through the source code and try different techniques to come up with a solution.

However, one thing you can do is enable the `basicAuth` guard during testing.

- Configure the auth package to enable the basic auth. You can do it by re-running the `node ace invoke @adonisjs/auth` command.
- Inside the routes file enabled the basic guard on the middleware
  
  ```ts
   Route.group(() => {

   }).middleware('auth:session,basic')
  ```
- Now inside your test you just need to set the `Authorization` header with the following `credentials` value.

  ```ts
   import { base64 } from '@poppinss/utils'
   const credentials = `Basic ${base64.encode('email:plaintextpassword')}`
   ```
